docs/tutorial_build_automation.rst

```diff
@@ -0,0 +1,174 @@
+For this reason, currently user must specify an environment.
+ReFrame treats Spack environments as *test resources* so it expects them under the test's :attr:`~reframe.core.pipeline.RegressionTest.sourcesdir`, which defaults to ``'src'``.
+Here is the directory structure for the test in this particular example that we show here:
+
+
+
+.. code:: console
+
+   tutorials/build_systems/spack/
+   ├── spack_test.py
+   └── src
+       └── myenv
+           └── spack.yaml
+
+
+We could have placed ``spack.yaml`` directly under the ``src/`` directory, in which case we would need to specify ``'.'`` as an environment.
+
+As with every other test, ReFrame will copy the test's resources to its stage directory before building it.
+ReFrame will then activate the environment and install the associated specs as in this case.
+Optionally, we can add more specs to the environment by setting the :attr:`~reframe.core.buildsystems.Spack.specs` attribute of the build system.
+Here is what ReFrame generates as a build script in this example:
+
+.. code:: bash
+
+   . $SPACK_ROOT/share/spack/setup-env.sh
+   spack env activate -V -d myenv
+   spack install
+
+Any additional specs specified inside the ReFrame test will be added using the ``spack add`` command.
+As you might have noticed ReFrame expects that Spack is already installed on the system.
+The packages specified in the environment and the tests will be installed in the test's stage directory, where the environment is copied before building.
+Here is the the stage directory structure:
+
+.. code:: console
+
+   stage/generic/default/builtin/BZip2SpackCheck/
+   ├── myenv
+   │   ├── spack
+   │   │   ├── opt
+   │   │   │   └── spack
+   │   │   │       ├── bin
+   │   │   │       └── darwin-catalina-skylake
+   │   │   └── share
+   │   │       └── spack
+   │   │           └── modules
+   │   ├── spack.lock
+   │   └── spack.yaml
+   ├── rfm_BZip2SpackCheck_build.err
+   ├── rfm_BZip2SpackCheck_build.out
+   ├── rfm_BZip2SpackCheck_build.sh
+   ├── rfm_BZip2SpackCheck_job.err
+   ├── rfm_BZip2SpackCheck_job.out
+   └── rfm_BZip2SpackCheck_job.sh
+
+
+Finally, here is the generated run script that ReFrame uses to run the test, once its build has succeeded:
+
+.. code-block:: bash
+
+   #!/bin/bash
+   . $SPACK_ROOT/share/spack/setup-env.sh
+   spack env activate -V -d myenv
+   bzip2 --help
+
+From this point on, sanity and performance checking are exactly identical to any other ReFrame test.
+
+.. tip::
+
+   While developing a test using Spack or EasyBuild as a build system, it's good to run ReFrame with :option:`--keep-stage-files` and :option:`--dont-restage` options.
+   These options will not remove the test stage files upon successful completion of the test but, more importantly, ReFrame will not wipe out the stage directory every time you retry the test, which is quite useful for avoiding rebuilding the dependencies of the packages built by the test.
```

Comment:
```suggestion
   While developing a test using Spack or EasyBuild as a build system, it can be useful to run ReFrame with :option:`--keep-stage-files` and :option:`--dont-restage` options to prevent ReFrame from removing the test's stage directory upon successful completion of the test.
   For this particular type of test, these options will avoid having to rebuild the required package dependencies every time the test is retried.
```